var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -129,6 +129,7 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage, CompilerPackage):
         description="Compilers and runtime libraries to build",
     )
     variant("binutils", default=False, description="Build via binutils")
+    variant("mold", default=False, description="Use mold as the linker by default")
```

Comment:
Commenting as a reminder to revisit before merging. Maybe add
```suggestion
    variant("mold", default=False, description="Use mold as the linker by default", when="@12:")
```
or
```suggestion
    variant("mold", default=False, description="Use mold as the linker by default")
    conflicts("+mold", when="@:11")
```